R/mobr_boxplots.R

```diff
@@ -131,44 +143,59 @@ get_CI <- function(x, level = 0.95)
 #' inv_mob_in = make_mob_in(inv_comm, inv_plot_attr)
 #' inv_stats = get_mob_stats(inv_mob_in, 'group', nperm=100)
 #' plot(inv_stats, multipanel=TRUE)
-get_mob_stats = function(mob_in, group_var, n_min = 10, nperm = 1000) {
+get_mob_stats = function(mob_in, group_var, ref_group = NULL, n_min = 5, nperm = 100) {
    if (nperm < 1) 
        stop('Set nperm to a value greater than 1') 
    group_id  = factor(mob_in$env[, group_var])
    
+   # Create factor with just two levels: treatment / control
+   if (is.null(ref_group))
+      ref_group = levels(group_id)[1]
+   group_bin = factor(rep("control", times = length(group_id)),
+                      levels = c("control","treatment"))
+   group_bin[group_id != ref_group] <- "treatment"
+   
    # Sample-based statistics
    N_sample = rowSums(mob_in$comm)      # individuals in each sample
    S_sample = rowSums(mob_in$comm > 0)  # species in each sample
    
    # rarefied richness
-   N_min_sample = max(n_min, min(N_sample))
+   N_min_sample = min(N_sample)
+   rarefy_samples = c(n_min, floor(N_min_sample/2), N_min_sample)
+   rarefy_samples[rarefy_samples < n_min] <- n_min
    plots_low_n = N_sample < n_min
    
    if (sum(plots_low_n) > 0){
       warning(paste("There are",sum(plots_low_n),"plots with less then", n_min,"individuals. These plots are removed for the calculation of rarefied richness."))
    }
 
-   S_rare_sample = rep(NA, nrow(mob_in$comm))
-   S_rare_sample[!plots_low_n] = apply(mob_in$comm[!plots_low_n,], MARGIN = 1,
-                                       FUN = rarefaction, method = "indiv",
-                                       effort = N_min_sample)
+   S_rare = data.frame(S_rare1 = rep(NA, nrow(mob_in$comm)),
+                       S_rare2 = rep(NA, nrow(mob_in$comm)),
+                       S_rare3 = rep(NA, nrow(mob_in$comm)))
    
+   S_rare[!plots_low_n,] <-  t(apply(mob_in$comm[!plots_low_n,], MARGIN = 1,
+                                      FUN = rarefaction, method = "indiv",
+                                      effort = rarefy_samples))
+
    # Probability of Interspecific Encounter
-   plots_n0 = N_sample == 0
+   plots_n01 = N_sample <= 1 # Hurlbert's PIE can only be calculated for two or more individuals
    
-   if (sum(plots_n0) == 1){
-      warning(paste("There is",sum(plots_n0), "plot without any individuals. This is removed for the calculation of PIE."))
+   if (sum(plots_n01) == 1){
+      warning(paste("There is",sum(plots_n01), "plot with less than two individuals.
```

Comment:
So I still think we should give plots with one individual a PIE of zero rather than omitting them from the analysis. @rueuntal, @ngotelli, and @jmchase what do you guys think?